12_Teacher or Instructor involved.py

```diff
@@ -57,37 +57,45 @@ def courses_offered(courses, hexamester):
         res.append(c)
     return res
 
-# [Double Gold Star] Define a procedure, involved(courses, person), that takes 
-# as input a courses structure and a person and returns a Dictionary that 
-# describes all the courses the person is involved in.  A person is involved 
-# in a course if they are a value for any property for the course.  The output 
-# Dictionary should have hexamesters as its keys, and each value should be a 
-# list of courses that are offered that hexamester (the courses in the list 
+
+# [Double Gold Star] Define a procedure, involved(courses, person), that takes
+# as input a courses structure and a person and returns a Dictionary that
+# describes all the courses the person is involved in.  A person is involved
+# in a course if they are a value for any property for the course.  The output
+# Dictionary should have hexamesters as its keys, and each value should be a
+# list of courses that are offered that hexamester (the courses in the list
 # can be in any order).
 
 def involved(courses, person):
-    # Your Code here
+    dict = {}
 
+    for hexamester, hexacourse in courses.items():
+        for course_code, course in hexacourse.items():
+            for keys, values in course.items():

```

Comment:
Technically, variable `keys` will hold only one key at a time and so does 'values' hold the corresponding value of the key. The loop will be executed for each of the pair in the dict. So, we should singular names.